File: Frontend/MedSync-Frontend/src/app/Shared/Models/doctor-schedule.model.ts
export class DoctorSchedule {
    DoctorScheduleID?: number = 0;
    DoctorID: number = 0;
    DayOfWeek: number = 0;
    StartTime: string = '';
    EndTime: string = '';
    SlotDurationID: number = 0;
    IsActive: boolean = true;
    CreatedOn?: string = new Date().toISOString();
    CreatedBy?: number | null = null;
    ModifiedOn?: string | null = null;
    ModifiedBy?: number | null = null;
}

File: Backend/MedSync-API/MedSync-ClassLibraries/Models/DoctorScheduleModel.cs
using System;

namespace MedSync_ClassLibraries.Models
{
    public class DoctorScheduleModel
    {
        public int? DoctorScheduleID { get; set; }
        public int DoctorID { get; set; }
        public byte DayOfWeek { get; set; }
        public TimeSpan StartTime { get; set; }
        public TimeSpan EndTime { get; set; }
        public int SlotDurationID { get; set; }
        public bool IsActive { get; set; } = true;
        public DateTime CreatedOn { get; set; } = DateTime.Now;
        public int? CreatedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public int? ModifiedBy { get; set; }
    }
}
